dashboards/create-metrics.py

```diff
@@ -0,0 +1,110 @@
+import argparse
+import json
+import logging
+from google.cloud import logging as gcloud_logging
+
+
+def main(service_account_path, pre_defined_metrics_json='metrics_template.json', google_project=None):
+    logging.basicConfig(level=logging.INFO)
+
+    # Instantiate a client from the service account JSON key path
+    logging_client = gcloud_logging.Client.from_service_account_json(service_account_path)
+
+    # Load pre-defined metrics JSON
+    with open(pre_defined_metrics_json, 'r') as f:
+        METRICS = json.load(f)
+
+    # List the current metrics
+    list_metrics(logging_client)
+
+    # Create the necessary metrics from the pre-defined list
+    if google_project:
+        # Check if we are using the right service account JSON key
+        assert logging_client.project == google_project
+    google_project = logging_client.project
+
+    for metric in METRICS:
+        check_and_create_metrics(gcloud_log_client=logging_client, metric_json=metric, google_project=google_project)
+
+    # List the current metrics again
+    list_metrics(logging_client)
+
+    # Double check
+    assert len(METRICS) == len(list(logging_client.list_metrics()))
+
+
+def list_metrics(gcloud_log_client):
+    """List all of the log-based metrics that is currently available on the cloud.
+
+    Args:
+        gcloud_log_client (google.cloud.logging.client.Client): A concrete google cloud logging client instance.
+    """
+    logging.info("Listing all of the current metrics:")
+    for metric in gcloud_log_client.list_metrics():
+        print(metric.name)
+
+
+def check_and_create_metrics(gcloud_log_client, metric_json, google_project):
+    """Check the existence of a log-based metric, if it's not existing yet, create the metric.
+
+    Args:
+        gcloud_log_client (google.cloud.logging.client.Client): A concrete google cloud logging client instance.
+        metric_json (dict): A dict representing a parameterized log-based metric. e.g.
+            {
+                "NAME": "Falcon-igniter-effective-release",
+                "FILTER": " resource.type="container" logName="projects/{google_project}/logs/falcon" "
+            }
+        google_project (str): A string representing the google cloud project name.
+    """
+    logging.info("Loading the metric {}".format(metric_json))
+
+    # The following line would not fail even if the metric_json is not parameterized :)
```

Comment:
I'm not sure I understand this comment fully...can you do a try/catch to check that the metric is correct?

Reply:
Sorry for the confusion, this line means that both `"FILTER": "\nresource.type=\"container\"\nlogName=\"projects/{google_project}/logs/lira\"\n\"INFO:lira.api.notifications:Notification received\"\n"` and `"FILTER": "\nresource.type=\"container\"\nlogName=\"projects/project_name/logs/lira\"\n\"INFO:lira.api.notifications:Notification received\"\n"` could work, based on the Python machinery.

Reply:
Updated the comment to avoid further confusion!